### PR TITLE
quiche: update tar ball to f555d99a084cdd086a349548c70fb558ac5847cf

### LINKED ISSUE #10420: MSVC Compilation errors with current quiche implementation
*Title*: *We encounter two different compilation errors building the quiche dependency on the MSVC compiler for Windows*

*Description*:
These two upstream errors prevent us from building the quic_listeners tests on Windows;

This first compilation failure should be relatively straightforward to review and correct;
```
ERROR: C:/_eb/external/com_googlesource_quiche/BUILD.bazel:2762:1: C++ compilation of rule '@com_googlesource_quiche//:quic_core_sent_packet_manager_lib' failed (Exit 2)
C:\_eb\execroot\envoy\bazel-out\x64_windows-opt\bin\external\com_googlesource_quiche\quiche\quic\core\quic_sent_packet_manager.cc(754) : error C2220: the following warning is treated as an error
C:\_eb\execroot\envoy\bazel-out\x64_windows-opt\bin\external\com_googlesource_quiche\quiche\quic\core\quic_sent_packet_manager.cc(754) : warning C4715: 'quic::QuicSentPacketManager::OnRetransmissionTimeout': not all control paths return a value
Target //test/extensions/quic_listeners/quiche:envoy_quic_dispatcher_test failed to build
```

This second compilation failure seems similar to other differences of interpretation and inferences drawn by gcc, clang and the Microsoft VC compiler, and we have been unable to sort it out;
```
ERROR: C:/_eb/external/com_googlesource_quiche/BUILD.bazel:2941:1: C++ compilation of rule '@com_googlesource_quiche//:quic_core_stream_send_buffer_lib' failed (Exit 2)
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.24.28314\include\xutility(4853): error C2678: binary '*': no operator found which takes a left-hand operand of type 'const _InIt' (or there is no acceptable conversion)
        with
        [
            _InIt=quic::QuicIntervalDeque<quic::BufferedSlice,quic::QuicCircularDeque<quic::BufferedSlice,3,std::allocator<quic::BufferedSlice>>>::Iterator
        ]
bazel-out/x64_windows-opt/bin/external/com_googlesource_quiche\quiche/quic/core/quic_interval_deque.h(188): note: could be 'quic::BufferedSlice &quic::QuicIntervalDeque<quic::BufferedSlice,quic::QuicCircularDeque<T,3,std::allocator<T>>>::Iterator::operator *(void)'
        with
        [
            T=quic::BufferedSlice
        ]
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.24.28314\include\xutility(4853): note: while trying to match the argument list '(const _InIt)'
        with
        [
            _InIt=quic::QuicIntervalDeque<quic::BufferedSlice,quic::QuicCircularDeque<quic::BufferedSlice,3,std::allocator<quic::BufferedSlice>>>::Iterator
        ]
bazel-out/x64_windows-opt/bin/external/com_googlesource_quiche/quiche/quic/core/quic_stream_send_buffer.cc(240): note: see reference to function template instantiation '_FwdIt std::lower_bound<quic::QuicIntervalDeque<quic::BufferedSlice,quic::QuicCircularDeque<T,3,std:
:allocator<T>>>::Iterator,quic::QuicStreamOffset,quic::`anonymous-namespace'::CompareOffset>(_FwdIt,const _FwdIt,const _Ty &,_Pr)' being compiled
        with
        [
            _FwdIt=quic::QuicIntervalDeque<quic::BufferedSlice,quic::QuicCircularDeque<quic::BufferedSlice,3,std::allocator<quic::BufferedSlice>>>::Iterator,
            T=quic::BufferedSlice,
            _Ty=quic::QuicStreamOffset,
            _Pr=quic::`anonymous-namespace'::CompareOffset
        ]
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.24.28314\include\xutility(4853): error C2100: illegal indirection
```

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Update QUICHE to commit f555d99a084cdd086a349548c70fb558ac5847cf to fix some Windows compilation errors.

Remove EnvoyCryptoServerStream interface because it was there purely to accessing proof source in quic:: QuicCryptoServerStream and quic:: TlsServerHandshaker. Now that QUICHE provide a common interface for these two to expose ProofSource object, we don't need to override them to obtain and access the proof source any more. Remove the envoy subclasses EnvoyQuicCryptoServerStream and EnvoyQuicTlsServerHandshaker.

Fixes #10420, #13154

### CODE REVIEW

File: bazel/external/quiche.BUILD
Comment (wu-bin) [inline bazel/external/quiche.BUILD:62]: nit: Fix indent

**Review by wu-bin — DISMISSED** [review]
Thanks Dan, LGTM.

File: bazel/external/quiche.BUILD
Comment (danzh2010) [reply bazel/external/quiche.BUILD]: done

**Review by ggreenway — DISMISSED** [review]
LGTM. Thanks!

**Comment by ggreenway** [comment]
@danzh2010 there's a gcc CI failure in quiche (not envoy) code.

/wait

**Comment by ggreenway** [comment]
gcc is still broken in quiche

/wait

**Comment by wrowe** [comment]
This looks good now on gcc @ggreenway

@danzh2010 TYVM!

**Comment by danzh2010** [comment]
Had to fix more in upstream, PTAL

**Review by wu-bin — APPROVED** [review]
Still LGTM, thanks Dan!

**Comment by wrowe** [comment]
As this is a blocking issue for other PR's we would ask @envoyproxy/senior-maintainers if anyone has cycles to give this a green light and pull it?

**Review by ggreenway — APPROVED** [review]
LGTM. Thanks for making all the fixes.

**Pull request merged**
